File: Strings/__2000__/2728/2728.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner in = new Scanner(System.in);
        String cobol = "cobol";

        while (in.hasNextLine()) {
            String[] palavras = in.nextLine().split("-");
            boolean valido = true;

            for (int i = 0; i < 5; i++) {
                String curr = palavras[i].toLowerCase();
                String letra = String.valueOf(cobol.charAt(i));

                if (!curr.startsWith(letra) && !curr.endsWith(letra)) {
                    valido = false;
                    break;
                }
            }

            System.out.println(valido ? "GRACE HOPPER" : "BUG");
        }

        in.close();
    }
}

File: Strings/__2000__/2728/2728.js
const fs = require("fs")
const input = fs.readFileSync("/dev/stdin", "utf8").split("\n")

const cobol = "cobol"

input.pop() //nao sei porque

for(const line of input){
    const palavras = line.split("-")

    let valido = true

    let ind = 0
    for(const palavra of palavras){
        let letra = cobol[ind]
        let curr = palavra.toLowerCase()

        if(!curr.startsWith(letra) && !curr.endsWith(letra)){
            valido = false;
            break;
        }
        ind++
    }

    console.log(valido ? "GRACE HOPPER": "BUG")
}